src/main/java/ru/rt/restream/reindexer/convert/FieldConverterRegistryFactory.java

```diff
@@ -0,0 +1,202 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ru.rt.restream.reindexer.convert;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.rt.restream.reindexer.annotations.Convert;
+import ru.rt.restream.reindexer.convert.util.ConversionUtils;
+import ru.rt.restream.reindexer.convert.util.ResolvableType;
+import ru.rt.restream.reindexer.util.Pair;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * For internal use only, as this contract is likely to change.
+ */
+public enum FieldConverterRegistryFactory implements FieldConverterRegistry {
+    INSTANCE;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FieldConverterRegistryFactory.class);
+
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+    private final Map<Class<?>, GenericFieldConverter<?, ?>> converters = new HashMap<>();
+
+    private final Map<ResolvableType, GenericFieldConverter<?, ?>> globalConverters = new HashMap<>();
+
+    private final Map<Pair<Class<?>, String>, GenericFieldConverter<?, ?>> fieldConverters = new HashMap<>();
+
+    /**
+     * Returns a {@link GenericFieldConverter} that is mapped for the given {@code field} or {@literal null}.
+     * The {@code X} and {@code Y} parameters are resolved and accessible via {@code GenericFieldConverter#getConvertiblePair}. 
+     * If the {@code FieldConverter} is not mapped directly for the field, looks up for a global one registered via
+     * {@link FieldConverterRegistry#registerGlobalConverter(FieldConverter)}, if global {@code FieldConverter} is found checks
+     * the {@code converterClass} from {@link Convert} annotation if its type is not default and does not match
+     * the global converter's type, then specified in the annotation converter takes precedence and
+     * is created for the {@code field}.
+     * This method returns {@literal null} if {@code Convert} annotation's attribute {@code disableConversion} is
+     * set to {@literal true} or if none of the checks above resulted in finding an eligible field converter. 
+     * @param field the {@link Field} to use
+     * @return the {@link GenericFieldConverter} to use
+     */
+    @SuppressWarnings("unchecked")
+    public <X, Y> GenericFieldConverter<X, Y> getFieldConverter(Field field) {
```

Comment:
This method both returns previously registered converters and registers converters for fields. Meanwhile, `registerFieldConverter()` is not used.
Is it possible to remove registration from `getFieldConverter()` and perform it during index parsing? At the same time, instead of two fields converters and globalConverters - get by with one?

Reply:
`registerFieldConverter` is supposed to be used to programatically add field converters, in case if an external library has its own mapping mechanism like the one Spring Data does `@ValueConverter`.